pkg/grapicmd/di/providers.go

```diff
@@ -70,23 +77,37 @@ func ProvideGexConfig(ctx *grapicmd.Ctx) *gex.Config {
 	return gexCfg
 }
 
-func ProvideInitializeProjectUsecase(ctx *grapicmd.Ctx, gexCfg *gex.Config, ui clui.UI, generator module.Generator) usecase.InitializeProjectUsecase {
-	return usecase.NewInitializeProjectUsecase(
-		ui,
-		generator,
-		gexCfg,
-		ctx.Version,
-	)
+func ProvideToolRepository(ctx *grapicmd.Ctx, gexCfg *gex.Config) (tool.Repository, error) {
+	toolRepoMu.Lock()
+	defer toolRepoMu.Unlock()
+	if toolRepo == nil {
+		var err error
+		toolRepo, err = gexCfg.Create()
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+	}
+	return toolRepo, nil
 }
 
-func ProvideExecuteProtocUsecase(ctx *grapicmd.Ctx, gexCfg *gex.Config, ui clui.UI, executor command.Executor, generator module.Generator) usecase.ExecuteProtocUsecase {
-	return usecase.NewExecuteProtocUsecase(
+func ProvideProtocWrapper(ctx *grapicmd.Ctx, ui clui.UI, toolRepo tool.Repository) protoc.Wrapper {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideProtocWrapper should have comment or be unexported